ldms/src/sampler/loadavg/loadavg.c

```diff
@@ -0,0 +1,438 @@
+	int i;
+	int k = 0;
+	for (i = 0; i < METLEN; i++)
+		if (metname[i].collect)
+			k += 1;
+	if (k == METLEN) {
+		return default_schema_name;
+	}
+	snprintf(der_schema, sizeof(der_schema), "loadavg%d%d%d%d%d%d",
+		metname[0].collect,
+		metname[1].collect,
+		metname[2].collect,
+		metname[3].collect,
+		metname[4].collect,
+		metname[5].collect
+	);
+	return der_schema;
+}
+
+static ldms_set_t get_set(struct ldmsd_sampler *self)
+{
+	return set;
+}
+
+static int create_metric_set(base_data_t base)
+{
+	ldms_schema_t schema;
+	int rc;
+	int i;
+
+	mf = fopen(procfile, "r");
+	if (!mf) {
+		msglog(LDMSD_LERROR, SAMP ": Could not open " PROC_FILE
+				" : exiting sampler\n");
+		return ENOENT;
+	}
+	fclose(mf);
+	mf = NULL;
+
+	schema = base_schema_new(base);
+	if (!schema) {
+		rc = ENOMEM;
+		goto err;
+	}
+
+	/* Location of first metric from proc file */
+	metric_offset = ldms_schema_metric_count_get(schema);
+
+
+	/* Make sure these are added in the order they will appear in the file */
+	for (i = 0; i < METLEN; i++) {
+		if (metname[i].collect) {
+			/*
+			msglog(LDMSD_LDEBUG, SAMP ": adding metric %s %d\n",
+				metname[i].name, (int)metname[i].vtype);
+			*/
+			/* rc = ldms_schema_metric_add(schema, metname[i].name, metname[i].vtype); */
+			/* all coerced to u64 for now... */
+			rc = ldms_schema_metric_add(schema, metname[i].name, LDMS_V_U64);
+			if (rc < 0) {
+				rc = ENOMEM;
+				goto err;
+			}
+		} else {
+			/*
+			msglog(LDMSD_LDEBUG, SAMP ": skipping metric %s\n",
+				metname[i].name);
+			*/
+		}
+	}
+	set = base_set_new(base);
+	if (!set) {
+		rc = errno;
+		goto err;
+	}
+
+	return 0;
+
+err:
+	schema = NULL;
+	return rc;
+}
+
+/**
+ * check for invalid flags, with particular emphasis on warning the user about
+ */
+static int config_check(struct attr_value_list *kwl, struct attr_value_list *avl, void *arg)
+{
+	char *value;
+	int i;
+
+	char* deprecated[]={"set"};
+
+	for (i = 0; i < (sizeof(deprecated)/sizeof(deprecated[0])); i++){
+		value = av_value(avl, deprecated[i]);
+		if (value){
+			msglog(LDMSD_LERROR, SAMP ": config argument %s has been deprecated.\n",
+			       deprecated[i]);
+			return EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static const char *usage(struct ldmsd_plugin *self)
+{
+	return  "config name=" SAMP BASE_CONFIG_USAGE " [metrics=<mlist>]\n"
+		"    <sname>      The schema name, if generated default is not good enough.\n"
+		"    <mlist>      comma separated list of metrics to include. If not given, all are included.\n"
+		"                 complete list is " MET1 "," MET2 "," MET3 "," MET4 "," MET5 "," MET6 "\n"
+		"\n";
+}
+
+/**
+ * \brief Configuration
+ *
+ * config name=loadavg producer_name=<name> instance_name=<instance_name> [component_id=<compid> schema=<sname>] [metrics=list]
+ *     producer_name    The producer id value.
+ *     instance_name    The set name.
+ *     component_id     The component id. Defaults to zero
+ *     sname            Optional schema name. Defaults to meminfo
```

Comment:
Could you please fix this copy and paste error? I think this is supposed to say 'loadavg'.